### ISSUE TITLE
How to calculate how much bytes were dowloaded from playback Buffer start event

### ISSUE THREAD

**Issue by giladna** [google/ExoPlayer#7101]
Hi,
Is there a way that we can calculate how much buffer was downloaded from Buffer start event till the Ready event?

so upon the bytes that were downloaded  by player and the minimum buffer for playback value we know we can predict when player will start playing by presenting this % of load to users

thanks!

**Comment by marcbaechinger** [comment]
I think that's very difficult to accurately calculate with the callbacks which are available. Certainly, `onLoadCompleted` comes to mind but that's not called frequent enough to give you the granularity you want.

In theory, the `EventTime` which is passed to every callback of the `AnalyticsListener` has a field  `totalBufferedDurationMs`. So you get the value of 'player.getTotalBufferedDuration()' with every event. However, I don't see what events would arrive between falling to buffered state until the player gets ready again.

You can poll 'player.getTotalBufferedDuration()', that's what the ExoPlayer UI components are doing.

**Comment by giladna** [comment]
hi @marcbaechinger ,

After checking the polling tip it seems that the buffer position is not something we can count on 
since it will have different value in each media load instance.

so we cannot assume that when first ready event is fired the buffer position will be around 2500 ms and that the load percentage can be calculated by that information

so even if the AnalyticsListener had such events it would not help me and 

* Is there some other option to figure out how long the buffering to ready period will take then?

This is snippet of what I did 
I 'v canceled the timer once ready was fired.

```
if (event.newState == PlayerState.BUFFERING) {
                    appProgressBar.setVisibility(View.VISIBLE);

                    timer = new Timer();
                    timer.scheduleAtFixedRate(new TimerTask() {
                        @Override
                        public void run() {
                            log.d("getBufferedPosition = " + player.getBufferedPosition());
                        }
                    }, 0, 100);
                }
```

PS.

One observation I saw that when calling player.getBufferedPercentage()
I always get 0 as the result 
worth checking the reason for this.

10x
Gilad.
